packages/@aws-cdk/aws-glue/lib/job.ts

```diff
@@ -606,7 +587,8 @@ export class Job extends JobBase {
   public static fromJobAttributes(scope: constructs.Construct, id: string, attrs: JobAttributes): IJob {
     class Import extends JobBase {
       public readonly jobName = attrs.jobName;
-      public readonly jobArn = JobBase.buildJobArn(scope, attrs.jobName);
+      public readonly jobArn = jobArn(scope, attrs.jobName)
+      public readonly grantPrincipal = new iam.UnknownPrincipal({ resource: this });
```

Comment:
fine for now, but you can also add this to `JobAttributes`

Reply:
add `role` to `JobAttributes` in https://github.com/aws/aws-cdk/pull/12506/commits/2ed79e1bc7cb557de4c166fd8bc70eaafe86f8b3